### ISSUE TITLE
Type checking: check schema.core value schemas (enum, maybe, either, primitives) at call sites

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2919]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

`schema.core` (Prismatic Schema) value schemas used with `:-` in `s/defn`/`s/fn`, or bound via `s/defschema`, are stripped by clj-kondo before type checking, so no mismatch is ever reported for them. For example:

```clojure
(require '[schema.core :as s])
(s/defn f [x :- s/Str] x)
(f 1)
```

This does not warn, even though `schema.core` itself would reject the call at runtime with `x` failing to satisfy `s/Str`. The same is true for `(s/enum ...)`, `(s/maybe ...)`, `(s/either ...)`, map schemas (`{...}`, `s/optional-key`) and sequential schemas (`[...]`), whether used inline or through a `s/defschema` name.

**Describe the solution you'd like**

Infer a clj-kondo type tag from these schema forms (primitives like `s/Str`/`s/Int`, `s/enum`, `s/maybe`, `s/either`, map schemas, sequential schemas, and same-file `s/defschema` names referring to any of the above) and feed it into the existing type checker, so mismatches at `s/defn`/`s/fn` param and return positions are reported the same way ordinary type-hinted params are.

**Describe alternatives you've considered**

Leaving schemas purely as documentation/runtime-only validation, as today. This misses static-analysis value that clj-kondo already provides for plain type hints, and schema usage is common enough (existing `s/defn`, `s/defrecord`, `s/defprotocol` support, misplaced-return-schema linting) that closing this gap seems worthwhile.

**Additional context**

I have a PR ready for this: #2918
